Not sure how to implement

This is exactly what I've been looking for but I'm not familiar with your development environment (just have been using Arduino).

Please reach out to me and let me know.

Just really looking for a turnkey implementation on my Atom.

Thank you!

Well, now that I figured out PlatformIO and how launch it into the app, the build/upload failed. Here's the terminal session log. Perhaps there's a missing or incorrectly named library? Also, I've loaded this in my Arduino library folder ... maybe it should be elsewhere on my drive?

Thanks for your insight!


> Executing task: C:\Users\1123781\.platformio\penv\Scripts\pio.exe run --environment m5stack-core-esp32 <

Processing m5stack-core-esp32 (platform: espressif32; board: m5stick-c; framework: arduino)
------------------------------------------------------------------------------------------------------
Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/m5stick-c.html
PLATFORM: Espressif 32 (2.0.0) > M5Stick-C
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
DEBUG: Current (esp-prog) External (esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES:
 - framework-arduinoespressif32 3.10004.201016 (1.0.4)
 - tool-esptoolpy 1.20600.0 (2.6.0)
 - toolchain-xtensa32 2.50200.80 (5.2.0)
LDF: Library Dependency Finder -> http://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 28 compatible libraries
Scanning dependencies...
Dependency Graph
|-- <M5Atom> 0.0.1
|   |-- <FastLED> 3.3.3
|   |   |-- <SPI> 1.0
|   |-- <Wire> 1.0.1
|-- <FastLED> 3.3.3
|   |-- <SPI> 1.0
|-- <ESPmDNS> 1.0
|   |-- <WiFi> 1.0
|-- <WiFi> 1.0
Building in release mode
Compiling .pio\build\m5stack-core-esp32\src\messageHandler.cpp.o
Archiving .pio\build\m5stack-core-esp32\libb94\libM5Atom.a
Compiling .pio\build\m5stack-core-esp32\lib500\WiFi\WiFiAP.cpp.o
src\messageHandler.cpp:19:28: fatal error: ESP32WebServer.h: No such file or directory

************************************************************************
* Looking for ESP32WebServer.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:ESP32WebServer.h"
* Web  > Compiling .pio\build\m5stack-core-esp32\lib500\WiFi\WiFiClient.cpp.o
https://platformio.org/lib/search?query=header:ESP32WebServer.h
*
************************************************************************

compilation terminated.
Compiling .pio\build\m5stack-core-esp32\lib500\WiFi\WiFiGeneric.cpp.o
Compiling .pio\build\m5stack-core-esp32\lib500\WiFi\WiFiMulti.cpp.o
Compiling .pio\build\m5stack-core-esp32\lib500\WiFi\WiFiSTA.cpp.o
Compiling .pio\build\m5stack-core-esp32\lib500\WiFi\WiFiScan.cpp.o
*** [.pio\build\m5stack-core-esp32\src\messageHandler.cpp.o] Error 1
===================================== [FAILED] Took 4.85 seconds =====================================
The terminal process "C:\Users\1123781\.platformio\penv\Scripts\pio.exe 'run', '--environment', 'm5stack-core-esp32'" terminated with exit code: 1.

Terminal will be reused by tasks, press any key to close it.

Getting closer, figured out library dependency except it seems esp32webserver. Is there a specifc version or source you're using? 

Here's the run now:


Processing m5stack-core-esp32 (platform: espressif32; board: m5stick-c; framework: arduino)
-------------------------------------------------------------------------------------------------------
Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/m5stick-c.html
PLATFORM: Espressif 32 (2.0.0) > M5Stick-C
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
DEBUG: Current (esp-prog) External (esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES:
 - framework-arduinoespressif32 3.10004.201016 (1.0.4)
 - tool-esptoolpy 1.20600.0 (2.6.0)
 - toolchain-xtensa32 2.50200.80 (5.2.0)
LDF: Library Dependency Finder -> http://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 28 compatible libraries
Scanning dependencies...
Dependency Graph
|-- <M5Atom> 0.0.1
|   |-- <FastLED> 3.3.3
|   |   |-- <SPI> 1.0
|   |-- <Wire> 1.0.1
|-- <FastLED> 3.3.3
|   |-- <SPI> 1.0
|-- <ESPmDNS> 1.0
|   |-- <WiFi> 1.0
|-- <WiFi> 1.0
Building in release mode
Compiling .pio\build\m5stack-core-esp32\src\messageHandler.cpp.o
src\messageHandler.cpp:19:28: fatal error: ESP32WebServer.h: No such file or directory

************************************************************************
* Looking for ESP32WebServer.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:ESP32WebServer.h"
* Web  > https://platformio.org/lib/search?query=header:ESP32WebServer.h
*
************************************************************************

compilation terminated.
*** [.pio\build\m5stack-core-esp32\src\messageHandler.cpp.o] Error 1
===================================== [FAILED] Took 2.99 seconds =====================================
The terminal process "C:\Users\1123781\.platformio\penv\Scripts\pio.exe 'run'" terminated with exit code: 1.

Terminal will be reused by tasks, press any key to close it.

--- a/.gitignore
+++ b/.gitignore
@@ -5,3 +5,4 @@
 .vscode/ipch
 .idea/
 .vscode/
+lib/

--- a/platformio.ini
+++ b/platformio.ini
@@ -15,7 +15,6 @@ framework = arduino
 upload_speed = 1500000
 monitor_speed = 115200
 lib_deps = 
-	m5stack/M5Atom @ ^0.0.1
 	m5stack/M5Atom @ ~0.0.1
-	m5stack/M5Atom @ 0.0.1
+	pedroalbuquerque/ESP32WebServer @ ^1.0
 	fastled/FastLED@3.3.3

